### PR TITLE
Add StopDaemon rpc.

### PR DESCRIPTION
**Pull request by sebdeveloper6952**
#### Pull Request Checklist

- [ x ] PR is opened against correct version branch.
- [ x ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [ x ] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

### CODE REVIEW

File: lightning_client.go [hunk 1]
```diff
@@ -3689,3 +3693,18 @@ func (s *lightningClient) RegisterRPCMiddleware(ctx context.Context,
 
 	return errChan, nil
 }
+
+func (s *lightningClient) StopDaemon(ctx context.Context) error {
```
Comment (guggero) [inline lightning_client.go:3697]: nit: add same Godoc comment here.

File: lightning_client.go [hunk 2]
```diff
@@ -3689,3 +3693,18 @@ func (s *lightningClient) RegisterRPCMiddleware(ctx context.Context,
...
+	rpcCtx, cancel := context.WithTimeout(ctx, s.timeout)
+	defer cancel()
+
+	rpcCtx = s.adminMac.WithMacaroonAuth(rpcCtx)
+	rpcReq := &lnrpc.StopRequest{}
+
+	_, err := s.client.StopDaemon(rpcCtx, rpcReq)
+	if err != nil {
```
Comment (guggero) [inline lightning_client.go:3705]: nit: return err directly.

**Review by guggero — CHANGES_REQUESTED** [review]
Thank you for the PR!
Since this RPC existed in `lnd` for a long time, I think we should open this against the `master` branch (currently tracking 0.13.x) and then rebase all other branches.

**Comment by sebdeveloper6952** [comment]
Thank you for the suggestions @guggero!

**Pull request closed without merging**